operator/README.md

```diff
@@ -52,26 +52,22 @@ eval $(minikube -p minikube docker-env)
 Compile the project and generate the Docker image with JIB:
 
 ```bash
-mvn clean package -Doperator -Dquarkus.container-image.build=true -Dquarkus.kubernetes.deployment-target=minikube
+mvn clean package -Doperator -Dquarkus.container-image.build=true -Dquarkus.kubernetes.namespace=keycloak -Dquarkus.kubernetes.annotations.\"olm.targetNamespaces\"=JOSDK_WATCH_CURRENT
```

Comment:
I think we should default to watching just the current namespace (without the need for additional properties during build). I.e. the generated K8s resources will be namespace scoped by default. This is the current behaviour.

For OLM, we should be able to interpret the `status.namespace` field on the CSV (mapped to some env var), incl. the special value for empty string (cluster-wide monitoring).

Worst case scenario, we can have some Maven profile for the two scenarios – watch current namespace by default; set some profile for honoring namespaces as set by OLM.

Reply:
The issue is that if we specify the namespaces to watch in the code with an annotation, then the override at run time does not work.

We need to specify the namespaces at build time with a property, then they can be overriden at runtime with an environment variable, which in the OLM case will support the empty string once [this issue](https://github.com/quarkiverse/quarkus-operator-sdk/issues/656) is fixed, which I have submitted a PR for.

Can we set build time properties in the maven profile? That might be the best way to do it.

Reply:
> The issue is that if we specify the namespaces to watch in the code with an annotation, then the override at run time does not work.

We don't want to be setting namespaces in code at all, do we?

> Can we set build time properties in the maven profile? That might be the best way to do it.

Yes, it should be possible to set them just via a system property.

Reply:
> Can we set build time properties in the maven profile? That might be the best way to do it.

Sure this can be changed to:

mvn clean package -Pbuild_single_ns -Dquarkus.kubernetes.namespace=keycloak

where the single_ns profile defaults to 
-Doperator -Dquarkus.container-image.build=true -Dquarkus.kubernetes.annotations.\"olm.targetNamespaces\"=JOSDK_WATCH_CURRENT

Reply:
I'd argue the `build_single_ns` profile should be active as default. :) It's what you usually want during development.

Reply:
Yep. Don't need to set quarkus.kubernetes.namespace - it can default to what's in the current context too.

Reply:
> We don't want to be setting namespaces in code at all, do we?

No we don't, absolutely not.

Reply:
> I'd argue the build_single_ns profile should be active as default. :) It's what you usually want during development.

You only need to use the profile to build the image AND deploy directly out of the target directory (either with apply or via quarkus).  Any other scenario and you won't need this profile.

> Yep. Don't need to set quarkus.kubernetes.namespace either - it can default to what's in the current context.

After the upstream changes this is true, but until then you either need this property to be set (so that the clusterrolebinding subjects have a namespace) or you have to use a josdk property to override the kubernetes.yml that is being generated.